src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java

```diff
@@ -61,43 +67,97 @@ public DriveBaseSubsystem(Joystick joystick) {
     // differential drive
     m_differentialDrive = new DifferentialDrive(m_motorControllers[Constants.driveLeftFrontIndex].getSparkMax(), 
                                           m_motorControllers[Constants.driveRightFrontIndex].getSparkMax());
+
+    this.usingExternal = usingExternal; // gets key if using external or internal encoders
+
+    // // external encoders            
+    // m_extLeftEncoder = new Encoder(Constants.leftEncoderDIOone, Constants.leftEncoderDIOtwo, 
+    //                          false, Encoder.EncodingType.k2X); // TODO: confirm correct configuration for encoder
+    // m_extRightEncoder = new Encoder(Constants.rightEncoderDIOone, Constants.rightEncoderDIOtwo, 
+    //                          false, Encoder.EncodingType.k2X);
+
+    // m_extLeftEncoder.setReverseDirection(true);
```

Comment:
If usingExternal gets true, you're code with crash because the m_ext*Encoders aren't initialized.